bundles/org.openhab.binding.network/src/main/java/org/openhab/binding/network/internal/WakeOnLanPacketSender.java

```diff
@@ -0,0 +1,127 @@
+/**
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.network.internal;
+
+import java.io.IOException;
+import java.net.DatagramPacket;
+import java.net.DatagramSocket;
+import java.net.InetAddress;
+import java.net.SocketException;
+import java.net.UnknownHostException;
+import java.util.Arrays;
+import java.util.Objects;
+import java.util.function.Consumer;
+import java.util.stream.Stream;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.smarthome.core.net.NetUtil;
+import org.eclipse.smarthome.core.util.HexUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link WakeOnLanPacketSender} broadcasts a magic packet to wake a device.
+ *
+ * @author Wouter Born - Initial contribution
+ */
+@NonNullByDefault
+public class WakeOnLanPacketSender {
+
+    private static final int WOL_UDP_PORT = 9;
+
+    // Wake-on-LAN magic packet constants
+    static final int PREFIX_BYTE_SIZE = 6;
+    static final int MAC_REPETITIONS = 16;
+    static final int MAC_BYTE_SIZE = 6;
+    static final int MAGIC_PACKET_BYTE_SIZE = PREFIX_BYTE_SIZE + MAC_REPETITIONS * MAC_BYTE_SIZE;
+    static final String[] MAC_SEPARATORS = new String[] { ":", "-" };
+
+    private final Logger logger = LoggerFactory.getLogger(WakeOnLanPacketSender.class);
+
+    private final String macAddress;
+    private byte @Nullable [] magicPacket;
+    private Consumer<byte[]> magicPacketSender;
+
+    public WakeOnLanPacketSender(String macAddress) {
+        this.macAddress = macAddress;
+        this.magicPacketSender = this::broadcastMagicPacket;
+    }
+
+    public WakeOnLanPacketSender(String macAddress, Consumer<byte[]> magicPacketSender) {
```

Comment:
This method is for testing only, right. So could be made package private?
```suggestion
    WakeOnLanPacketSender(String macAddress, Consumer<byte[]> magicPacketSender) {
```